main.js

```diff
@@ -86,35 +86,51 @@ selectSort.addEventListener('change', updateTable)
 
 const coursesTableBody = document.getElementById('coursesTableBody')
 
-function updateTable() {
+// Update the updateTable function to accept an optional searchQuery parameter
+function updateTable(searchQuery = '') {
   // Clear all existing rows
   coursesTableBody.innerHTML = ''
 
-  // Add a row to the table for each course
-  courses.forEach(course => {
+  // Filter courses based on the search query and other filters
+  const filteredCourses = courses.filter(course => {
+    // Check if the course name (DESCR) contains the search query
+    return (
+      course.CRSE.toLowerCase().includes(searchQuery.toLowerCase()) ||
+      course.DESCR.toLowerCase().includes(searchQuery.toLowerCase()) ||
+      course.INSTR.toLowerCase().includes(searchQuery.toLowerCase())
+    )
+  })
+
+  // Add a row to the table for each filtered course
+  filteredCourses.forEach(course => {
     const row = document.createElement('tr')
-    if (course.ENROLLING == 'Closed') row.classList.add('closed-course')
+    if (course.ENROLLING === 'Closed') row.classList.add('closed-course')
     columns.forEach(columnID => {
       const cell = document.createElement('td')
-      /* TODO: Filter based on current dropdown values (Issue #85)
-         The relevant element IDs are selectSubject, selectStartTime, and selectInstructionMode.
-         The select element's value will be blank if the "All" option is selected.
-         Otherwise, it the value will match the text displayed in the dropdown.
-      */
-      if (columnID == 'CONSENT') {
-        if (course.CONSENT == 'No Special Consent Required')
+      // Handle the CONSENT column separately as in your existing code
+      if (columnID === 'CONSENT') {
+        if (course.CONSENT === 'No Special Consent Required')
           cell.innerText = 'None'
-        else if (course.CONSENT == 'Instructor Consent Required')
+        else if (course.CONSENT === 'Instructor Consent Required')
           cell.innerText = 'Instructor'
-        else if (course.CONSENT == 'Department Consent Required')
+        else if (course.CONSENT === 'Department Consent Required')
           cell.innerText = 'Department'
         else cell.innerText = course[columnID]
-      } else cell.innerText = course[columnID]
+      } else {
+        cell.innerText = course[columnID]
+      }
       row.appendChild(cell)
     })
     coursesTableBody.appendChild(row)
   })
 }
 
+// Event listener for input changes in the search bar
+searchInput.setAttribute('size', searchInput.getAttribute('placeholder').length)
+searchInput.addEventListener('input', () => {
+  const searchQuery = searchInput.value
+  updateTable(searchQuery)
+})
+
 // Initial unfiltered table
 updateTable(courses, 'courseNumber')
```

Comment:
on the initial load up i think the 2 calls for update table or something are mixed up so we may want to create a new function called searchTable that will handle updating the table when searching because when we load it up at first nothing shows you have to search for it to show